app/lib/features/calendar_sync/providers/events_to_sync_provider.dart

```diff
@@ -0,0 +1,32 @@
+//ALL UPCOMING EVENTS
+import 'package:acter/features/events/actions/get_event_type.dart';
+import 'package:acter/features/events/providers/event_providers.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+typedef EventAndRsvp = ({CalendarEvent event, String? rsvp});
+
+final eventsToSyncProvider = FutureProvider.autoDispose((ref) async {
+  // fetch all from all spaces
+  final allEventList = await ref.watch(allEventListProvider(null).future);
+  final upcomingAndOngoing = allEventList.where((event) {
+    final eventType = getEventType(event);
+    return eventType == EventFilters.upcoming ||
+        eventType == EventFilters.ongoing;
+  });
+  final List<EventAndRsvp> toSync = [];
+
+  for (final event in upcomingAndOngoing) {
+    final eventId = event.eventId().toString();
+    final myRsvpStatus = await ref.watch(myRsvpStatusProvider(eventId).future);
+    final rsvpStatus = myRsvpStatus.statusStr();
+    if (rsvpStatus != 'no') {
+      // we sync all that aren't denied yet
+      final event = await ref.watch(
+        calendarEventProvider(eventId).future,
+      ); // ensure we are listening to updates of the events themselves
+      toSync.add((event: event, rsvp: rsvpStatus));
+    }
+  }
+  return toSync;
```

Comment:
How about using only single `for` loop?

```suggestion
   final List<EventAndRsvp> toSync = [];

  for (final event in allEventList) {
    final eventType = getEventType(event);
    if (eventType == EventFilters.upcoming ||
        eventType == EventFilters.ongoing) {
      final eventId = event.eventId().toString();
      final myRsvpStatus =
          await ref.watch(myRsvpStatusProvider(eventId).future);
      final rsvpStatus = myRsvpStatus.statusStr();
      if (rsvpStatus != 'no') {
        // we sync all that aren't denied yet
        final event = await ref.watch(
          calendarEventProvider(eventId).future,
        ); // ensure we are listening to updates of the events themselves
        toSync.add((event: event, rsvp: rsvpStatus));
      }
    }
  }
  return toSync;
```

Reply:
why? for code readability?

because the first is an iterator. in terms of execution this is _one_ for loop. And wherever possible I prefer to use iterators ... It's just that a) we are not sure about the behavior of ref.watch in iterator callbacks and b) it doesn't work if the result is an await/future at the moment ...